src/main/java/org/sagebionetworks/web/client/presenter/users/RegisterAccountPresenter.java

```diff
@@ -21,22 +21,21 @@ public void start(AcceptsOneWidget panel, EventBus eventBus) {}
 
   @Override
   public void setPlace(RegisterAccount place) {
-    String token = place.toToken();
-    if (token != null && token.contains("@")) {
-      // is likely an email address
-      Window.Location.replace(
-        WebConstants.ONESAGE_PRODUCTION_URL +
-        "/register1?email=" +
-        token +
-        "&" +
-        WebConstants.ONESAGE_SYNAPSE_APPID_QUERY_PARAM
-      );
-    } else {
-      Window.Location.replace(
-        WebConstants.ONESAGE_PRODUCTION_URL +
-        "/register1?" +
-        WebConstants.ONESAGE_SYNAPSE_APPID_QUERY_PARAM
-      );
+    String emailInvitationToken = place.getParam(
+      RegisterAccount.MEMBERSHIP_INVTN_QUERY_PARAM
+    );
+    String email = place.getParam(RegisterAccount.EMAIL_QUERY_PARAM);
+    StringBuilder targetUrl = new StringBuilder();
+    targetUrl.append(WebConstants.ONESAGE_PRODUCTION_URL);
+    targetUrl.append("/register1?");
+    targetUrl.append(WebConstants.ONESAGE_SYNAPSE_APPID_QUERY_PARAM);
+
+    if (emailInvitationToken != null) {
+      targetUrl.append("&signedToken=" + emailInvitationToken);
```

Comment:
This is where the magic happens.   One Sage already has logic to read a signed token from the query parameters and store it in Local Storage.  Then on the Account Created page, it already has logic to read this token in, check if it's of this type, and even automatically joins the target team (as well as changing the link to say something like "Take me to the Team"!).  This workflow functions perfectly (I just tested it), once we send over this signed token!